components/schemes/Color/src/MDCSemanticColorScheme.m

```diff
@@ -49,5 +49,19 @@ - (instancetype)initWithDefaults:(MDCColorSchemeDefaults)defaults {
   return self;
 }
 
++ (UIColor *)mergeColor:(UIColor *)color  withBackgroundColor:(UIColor *)backgroundColor {
+  CGFloat red = 0.0, green = 0.0, blue = 0.0, alpha = 0.0;
+  [color getRed:&red green:&green blue:&blue alpha:&alpha];
+  CGFloat bRed = 0.0, bGreen = 0.0, bBlue = 0.0, bAlpha = 0.0;
+  [backgroundColor getRed:&bRed green:&bGreen blue:&bBlue alpha:&bAlpha];
+
+  NSAssert(bAlpha == 1, @"Background color with opacity is not supported.");
```

Comment:
Please use a floating point equality check.
Why can't we support non-opaque background colors?
If this is a requirement, please note in the header.
In release code, this assert will be skipped, what is the fallback handling?